examples/QubitRotation.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Qubit Rotation with JAX
+# 
+# In this notebook, we shall investigate how we can use simple gradient descent to optimize for the a qubit rotation. The idea is as follows: We start in $|1 \rangle$ state on a Bloch Sphere and apply rotations around Z-Y-Z axes (in that order), each of the three with angles $\phi$, $\theta$, and $\omega$ respectively
+# 
+# $R(\phi,\theta,\omega) = RZ(\omega)RY(\theta)RZ(\phi)= \begin{bmatrix}
+# e^{-i(\phi+\omega)/2}\cos(\theta/2) & -e^{i(\phi-\omega)/2}\sin(\theta/2) \\
+# e^{-i(\phi-\omega)/2}\sin(\theta/2) & e^{i(\phi+\omega)/2}\cos(\theta/2)
+# \end{bmatrix}$
+# 
+# to reach the target state, which is defined to be $|0 \rangle$ in this case.
+
+
+import jax.numpy as jnp
```

Comment:
It is good practice to keep the imports from one library together leave a gap and import the other things in another block. Also import * is not recommended. Import specifically what you use in the code so that the user can clearly see where a particular function comes from. If you do from `qutip import *` and also from `qgrad import *` and use a function `fidelity()` then we do not know if it came from qutip or qgrad.

```
import jax.numpy as jnp
from jax import grad, jit
from jax.random import normal, PRNGKey


import numpy as onp


from qutip import fidelity, Qobj
```